src/Call.tsx

```diff
@@ -64,30 +64,20 @@ export declare interface Call {
 
   /** @internal */
   emit(
-    callEvent: Call.Event.QualityWarningsChanged,
+    qualityWarningsChangedEvent: Call.Event.QualityWarningsChanged,
     currentQualityWarnings: Call.QualityWarning[],
     previousQualityWarnings: Call.QualityWarning[]
   ): boolean;
 
+  /** @internal */
+  emit(callEvent: Call.Event, ...args: any[]): boolean;
+
```

Comment:
I'm thinking this one over again, and I think maybe we should remove the `emit(GenericEvent...` types, since the SDK is the one emitting and we never want to emit a generic event, unlike listening for a generic event. What do you think?

Reply:
@mhuynh5757 Actually, we should not have any emit type definitions at all, since the developer should not be emitting events on this object at all.

Reply:
I took a look at the TSDocs library documentation and it seems like there's no good solution here. The reason we wanted typed `.emit` declarations in the first place is for easier maintenance. If someone on the team tries to do something like `.emit(Call.Event.ConnectFailure)` and forget to pass the argument, then the compiler would fail and let us know we're emitting wrong.

It seems like marking the `.emit` as `@internal` is the proper solution, it signifies that 3rd party developers are not to use the API. But it doesn't hide it entirely, which is one downside. What do you think?

Reply:
Discussed with @manjeshbhargav over Slack, we've come to the conclusion that the `.emit` types should not hurt since they are marked `@internal` and that the `Generic` types also should not hurt since Intellisense will pick up on the other narrower types first.